common/src/main/java/org/opensearch/sdk/SdkClient.java

```diff
@@ -114,7 +124,12 @@ default DeleteDataObjectResponse deleteDataObject(DeleteDataObjectRequest reques
             if (cause instanceof InterruptedException) {
                 Thread.currentThread().interrupt();
             }
-            throw new OpenSearchException(cause);
+            // Rethrow unchecked Exceptions
+            if (cause instanceof RuntimeException) {
```

Comment:
`IllegalArgumentException` and `MLException` are both subclasses of `RuntimeException` so will be thrown here as the original exception.

`IOException` is a checked exception so is wrapped in an unchecked exception before throwing.

See the PR description where I suggested possibly wrapping in an `UndeclaredThrowableException` instead of `OpenSearchException`.

Do you have an alternative proposal here?  If we don't rethrow checked exceptions we need to add `throws` clauses to every one of these API methods.

Reply:
We have [MLException class](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/exception/MLException.java), may be we re-use them?

Reply:
These clients aren't unique to ML Commons.  The exceptions should be generic.  This class will not be in ML Commons eventually and shouldn't need to add it as a dependency.

The intent here is to rethrow the original exception whenever possible, and if not, pass a throwable with the original exception as its cause.